### PR TITLE
Issue #9480: updated example of AST for TokenTypes.LITERAL_IF

### LINKED ISSUE #9480: update example of AST for TokenTypes.LITERAL_IF
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_IF

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by theanmolsharma**
Fixes #9480 : 

Screenshot 2021-03-23 at 2 03 41 PM

Source used to generate AST:
```
public class MyClass {
    public static void main(String args[]) {
        String message = "";
        boolean optimistic = true;
        if(optimistic) {
            message = "half full";
        }
        else {
            message = "half empty";
        }
    }
}
```

Generated AST:
```
CLASS_DEF -> CLASS_DEF [9:0]
|--MODIFIERS -> MODIFIERS [9:0]
|   `--LITERAL_PUBLIC -> public [9:0]
|--LITERAL_CLASS -> class [9:7]
|--IDENT -> MyClass [9:13]
`--OBJBLOCK -> OBJBLOCK [9:21]
    |--LCURLY -> { [9:21]
    |--METHOD_DEF -> METHOD_DEF [10:4]
    |   |--MODIFIERS -> MODIFIERS [10:4]
    |   |   |--LITERAL_PUBLIC -> public [10:4]
    |   |   `--LITERAL_STATIC -> static [10:11]
    |   |--TYPE -> TYPE [10:18]
    |   |   `--LITERAL_VOID -> void [10:18]
    |   |--IDENT -> main [10:23]
    |   |--LPAREN -> ( [10:27]
    |   |--PARAMETERS -> PARAMETERS [10:39]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [10:39]
    |   |       |--MODIFIERS -> MODIFIERS [10:39]
    |   |       |--TYPE -> TYPE [10:39]
    |   |       |   `--ARRAY_DECLARATOR -> [ [10:39]
    |   |       |       |--IDENT -> String [10:28]
    |   |       |       `--RBRACK -> ] [10:40]
    |   |       `--IDENT -> args [10:35]
    |   |--RPAREN -> ) [10:41]
    |   `--SLIST -> { [10:43]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [11:8]
    |       |   |--MODIFIERS -> MODIFIERS [11:8]
    |       |   |--TYPE -> TYPE [11:8]
    |       |   |   `--IDENT -> String [11:8]
    |       |   |--IDENT -> message [11:15]
    |       |   `--ASSIGN -> = [11:23]
    |       |       `--EXPR -> EXPR [11:25]
    |       |           `--STRING_LITERAL -> "" [11:25]
    |       |--SEMI -> ; [11:27]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [12:8]
    |       |   |--MODIFIERS -> MODIFIERS [12:8]
    |       |   |--TYPE -> TYPE [12:8]
    |       |   |   `--LITERAL_BOOLEAN -> boolean [12:8]
    |       |   |--IDENT -> optimistic [12:16]
    |       |   `--ASSIGN -> = [12:27]
    |       |       `--EXPR -> EXPR [12:29]
    |       |           `--LITERAL_TRUE -> true [12:29]
    |       |--SEMI -> ; [12:33]
    |       |--LITERAL_IF -> if [13:8]
    |       |   |--LPAREN -> ( [13:10]
    |       |   |--EXPR -> EXPR [13:11]
    |       |   |   `--IDENT -> optimistic [13:11]
    |       |   |--RPAREN -> ) [13:21]
    |       |   |--SLIST -> { [13:23]
    |       |   |   |--EXPR -> EXPR [14:20]
    |       |   |   |   `--ASSIGN -> = [14:20]
    |       |   |   |       |--IDENT -> message [14:12]
    |       |   |   |       `--STRING_LITERAL -> "half full" [14:22]
    |       |   |   |--SEMI -> ; [14:33]
    |       |   |   `--RCURLY -> } [15:8]
    |       |   `--LITERAL_ELSE -> else [16:8]
    |       |       `--SLIST -> { [16:13]
    |       |           |--EXPR -> EXPR [17:20]
    |       |           |   `--ASSIGN -> = [17:20]
    |       |           |       |--IDENT -> message [17:12]
    |       |           |       `--STRING_LITERAL -> "half empty" [17:22]
    |       |           |--SEMI -> ; [17:34]
    |       |           `--RCURLY -> } [18:8]
    |       `--RCURLY -> } [19:4]
    `--RCURLY -> } [20:0]
```

Expected Update for Javadoc:
```
LITERAL_IF -> if
   |--LPAREN -> (
   |--EXPR -> EXPR
   |   `--IDENT -> optimistic
   |--RPAREN -> )
   |--SLIST -> {
   |   |--EXPR -> EXPR
   |   |   `--ASSIGN -> =
   |   |       |--IDENT -> message
   |   |       `--STRING_LITERAL -> "half full"
   |   |--SEMI -> ;
   |   `--RCURLY -> }
   `--LITERAL_ELSE -> else
       `--SLIST -> {
           |--EXPR -> EXPR
           |   `--ASSIGN -> =
           |       |--IDENT -> message
           |       `--STRING_LITERAL -> "half empty"
           |--SEMI -> ;
           `--RCURLY -> }
```

### CODE REVIEW

**Comment by timurt** [comment]
@theanmolsharma please place whole ast tree under `Generated AST:`

**Comment by theanmolsharma** [comment]
@timurt I updated the comment.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -1560,35 +1560,26 @@
...
-     *             |
-     *             +--EXPR
-     *                 |
-     *                 +--ASSIGN (=)
-     *                     |
-     *                     +--IDENT (message)
-     *                     +--STRING_LITERAL ("half empty")
-     *             +--SEMI (;)
-     *             +--RCURLY (})
+     * LITERAL_IF -&gt; if
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1563]: I think it should be `--LITERAL_IF` and next lines should have additional indents

I took as en example [TokenTypes#METHOD_DEF](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java#L198)

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -1560,35 +1560,26 @@
      * </pre>
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1560]: could you please fix example while you are on this token type, please put space after `if`

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (theanmolsharma) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1563]: I have worked on similar issues and I was always asked to remove additional indents by @pbludov , I can additional indents if you want.

**Review by theanmolsharma — COMMENTED** [review]
@timurt I have made the changes you requested. Please review whenever you have time.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: We have come to an agreement to keep `--`.  So it should start with `--LITERAL_IF`

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Let's remove "--".
All already merged cases can be fixed in separate issue.

Comment (theanmolsharma) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1563]: I have removed "--" and all additional indents.

**Review by timurt — APPROVED** [review]

**Review by romani — APPROVED** [review]

**Pull request merged**